pkg/internal/cluster/loadbalancer/config.go

```diff
@@ -33,18 +33,21 @@ type ConfigData struct {
 // DefaultConfigTemplate is the loadbalancer config template
 const DefaultConfigTemplate = `# generated by kind
 global
+  stats socket /var/run/haproxy.sock mode 600 level admin
+  stats timeout 2m
   log /dev/log local0
   log /dev/log local1 notice
   daemon
 
 defaults
   log global
   mode tcp
+  option tcpka
   option dontlognull
   # TODO: tune these
   timeout connect 5000
-  timeout client 50000
```

Comment:
what unit where these in?

Reply:
milliseconds https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#4-timeout%20server

Reply:
So 50s to an hour.

Seems arbitrarily large, how do other setups compare (eg kubespray)?

Reply:
is to be safe, maybe exaggerated, but don't think that anybody is going to exhaust the host with open tcp connections with kind :)

Reply:
fair I supposed, I'm not sure this is the correct or right fix, but I'm OK to try it, mostly seems harmless

Reply:
mild concern that we should not do non-production ready configs where avoidable.

I feel like the actual bug here is in whatever is talking to the API server without any data flowing or any reconnects, I'm not a fan of masking a bug there.

Reply:
is hard to say if it is a bug or a limitation, the software does what's supposed to do, just doesn`t implement keepalives to avoid the intermediate devices closes the connection, all of the involved components can be blamed, is a common TCP problem with long lived connections :man_shrugging:

Reply:
@BenTheElder I am going to finish [this patch](https://github.com/kubernetes/kubernetes/compare/master...ereslibre:ping-on-long-running-commands) and open a PR. It covers `kubectl exec -it`, `kubectl logs -f` and `kubectl get -w`. The idea is to ping the other end on a regular basis.

Reply:
thanks @ereslibre! + 💯

Reply:
The patch is almost ready. I will review it tomorrow, checking that I am not breaking anything, rereading for minor changes and I think I should be good to open a PR and ping all stakeholders.

Reply:
Side note, when testing manually I go to the extreme and use this patch in kind: https://github.com/kubernetes-sigs/kind/compare/master...ereslibre:low-timeout, it's much more convenient to force the issue to ensure that pings are working as intended.